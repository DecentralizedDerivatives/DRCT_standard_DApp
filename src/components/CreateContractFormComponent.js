import React from 'react';
import { Field } from 'redux-form';
import { InputGroup } from 'reactstrap';
import InputNumber from './InputNumber';
import Select from './Select.js';
import { factories } from '../factoryProvider';

export const CreateContractFormComponent = ({
  handleSubmit,
  onSubmit,
  contractDates,
  onCurrencyChange
}) => {
  let factoryList = factories().reduce((obj, item) => {
    obj[item.address] = item.symbol + ' (' + item.multiplier + 'X multiplier)';
    return obj
  }, {})
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <InputGroup className="form-dropdown">
        <Field
          name="duration"
          label="Select Duration"
          component={Select}
          options={{
            'One Week': 'One Week'
          }}
        />
      </InputGroup>

      <InputGroup className="form-dropdown">
        <Field
          name="currency"
          label="Select Underlying"
          component={Select}
          options={factoryList}
          onChange={onCurrencyChange}
        />
      </InputGroup>
      <InputGroup className="form-dropdown">
        <Field
          name="startDate"
          label="Start Date"
          component={Select}
          options={contractDates}
        />
      </InputGroup>
          <p style={{margin:'5px'}}>Enter the amount of Ether for the total purchase of the contract below.</p>
      <InputGroup className="form-eth-input">
        <Field name="amount" addonLabel="ETH" step="0.01" component={InputNumber} />
      </InputGroup>

      <div className="form-submit-wrapper">
        <button type="submit" className="form-submit btn btn-primary">
          Create Contract
        </button>
      </div>
    </form>
  );
};

export default CreateContractFormComponent;
