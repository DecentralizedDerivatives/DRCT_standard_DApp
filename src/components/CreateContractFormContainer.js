import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { reduxForm, getFormValues } from 'redux-form';
import { sendCreateContractOrder } from '../actions/orderActions';
import { getContractOpenDates } from '../actions/contractActions';
import CreateContractFormComponent from './CreateContractFormComponent';

const validate = values => {
  const errors = {};
  if (!values.duration) {
    errors.duration = 'Required';
  }

  if (!values.currency) {
    errors.currency = 'Required';
  }

  if (!values.startDate) {
    errors.startDate = 'Required';
  }

  if (!values.amount) {
    errors.amount = 'Required';
  }

  return errors;
};

export let CreateContractFormContainer = props => {
  const submitForm = formValues => {
    // console.log('submitting Form: ', formValues);
    props.sendCreateContractOrder(formValues, props.userAccount);
  };

  const onCurrencyChange = async evt => {
    var address = evt.target.value;
    await props.getContractOpenDates(address);
  };

  return (
    <div>
      <CreateContractFormComponent
        formValues={props.formValues}
        contractDates={props.contractDates}
        onCurrencyChange={onCurrencyChange}
        onSubmit={submitForm}
        handleSubmit={props.handleSubmit} />
    </div>
  );
};

CreateContractFormContainer.propTypes = {
  sendCreateContractOrder: PropTypes.func.isRequired,
  formValues: PropTypes.object,
  userAccount: PropTypes.string
};

const mapStateToProps = state => ({
  formValues: getFormValues('create-contract')(state),
  userAccount: state.user.userAccount,
  contractDates: state.contract.contractOpenDates
});

const formConfiguration = {
  form: 'create-contract',
  validate
};

CreateContractFormContainer = reduxForm(formConfiguration)(
  CreateContractFormContainer
);

export default (CreateContractFormContainer = connect(
  mapStateToProps,
  { getContractOpenDates, sendCreateContractOrder }
)(CreateContractFormContainer));
