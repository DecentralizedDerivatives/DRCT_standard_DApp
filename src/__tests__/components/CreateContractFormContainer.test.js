import CreateContractFormContainer from '../../components/CreateContractFormContainer';
import CreateContractFormComponent from '../../components/CreateContractFormComponent';

describe('<CreateContractFormContainer />', () => {
  describe('render()', () => {
    it('renders the component', async () => {
      const wrapper = shallow(
        <CreateContractFormContainer store={initStore()} />
      ).dive().dive().dive().dive();

      wrapper.find(CreateContractFormComponent).simulate('submit');

      expect(wrapper).toMatchSnapshot();
    });
  });
});
