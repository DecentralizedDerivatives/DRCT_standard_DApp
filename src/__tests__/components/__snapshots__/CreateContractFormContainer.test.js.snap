// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateContractFormContainer /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContractFormContainer
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    asyncValidating={false}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    clearAsyncError={[Function]}
    clearFields={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    contractDates={Object {}}
    destroy={[Function]}
    dirty={false}
    dispatch={[Function]}
    form="create-contract"
    getContractOpenDates={[Function]}
    handleSubmit={[Function]}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    pristine={true}
    pure={true}
    reset={[Function]}
    resetSection={[Function]}
    sendCreateContractOrder={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    submit={[Function]}
    submitFailed={false}
    submitSucceeded={false}
    submitting={false}
    touch={[Function]}
    untouch={[Function]}
    userAccount=""
    valid={true}
    validate={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CreateContractFormComponent
          contractDates={Object {}}
          handleSubmit={[Function]}
          onCurrencyChange={[Function]}
          onSubmit={[Function]}
        />,
        <div
          className="clickable"
          style={
            Object {
              "color": "blue",
              "marginTop": "20px",
              "textDecoration": "underline",
            }
          }
        >
          Already have a created contract?
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "contractDates": Object {},
          "formValues": undefined,
          "handleSubmit": [Function],
          "onCurrencyChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Already have a created contract?",
          "className": "clickable",
          "onClick": undefined,
          "style": Object {
            "color": "blue",
            "marginTop": "20px",
            "textDecoration": "underline",
          },
        },
        "ref": null,
        "rendered": "Already have a created contract?",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CreateContractFormComponent
            contractDates={Object {}}
            handleSubmit={[Function]}
            onCurrencyChange={[Function]}
            onSubmit={[Function]}
          />,
          <div
            className="clickable"
            style={
              Object {
                "color": "blue",
                "marginTop": "20px",
                "textDecoration": "underline",
              }
            }
          >
            Already have a created contract?
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "contractDates": Object {},
            "formValues": undefined,
            "handleSubmit": [Function],
            "onCurrencyChange": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Already have a created contract?",
            "className": "clickable",
            "onClick": undefined,
            "style": Object {
              "color": "blue",
              "marginTop": "20px",
              "textDecoration": "underline",
            },
          },
          "ref": null,
          "rendered": "Already have a created contract?",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
